src/Analyzer.Cli/CommandLineParser.cs

```diff
@@ -198,10 +198,56 @@ private void SetupCommonOptionsForCommands(List<Command> commands)
                 return (int)ExitCode.ErrorInvalidARMTemplate;
             }
 
-            var analysisResult = AnalyzeTemplate(templateFilePath, parametersFilePath);
+            ExitCode exitCode;
+            if (parametersFilePath != null)
+            {
+                exitCode = AnalyzeTemplate(templateFilePath, parametersFilePath);
+            }
+            else
+            {
+                // Check if a parameters.json file is present according to naming standards here https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/parameter-files#file-name, and if so use it as the parametersFile input
+                var parametersFiles = FindParameterFileForTemplate(templateFilePath);
+                int numOfFilesAnalyzed = 0;
+                bool issueReported = false;
+                var filesFailed = new List<FileInfo>();
+
+                if (parametersFiles.Count() > 0)
+                {
+                    foreach (FileInfo parametersFile in parametersFiles)
+                    {
+                        ExitCode res = AnalyzeTemplate(templateFilePath, parametersFile);
+
+                        if (res == ExitCode.Success || res == ExitCode.Violation)
+                        {
+                            numOfFilesAnalyzed++;
+                            issueReported |= res == ExitCode.Violation;
+                        }
+                        else if (res == ExitCode.ErrorAnalysis || res == ExitCode.ErrorInvalidBicepTemplate)
+                        {
+                            filesFailed.Add(templateFilePath);
+                        }
+                    }
+
+                    Console.WriteLine(Environment.NewLine + $"Analyzed {numOfFilesAnalyzed} {(numOfFilesAnalyzed == 1 ? "file" : "files")}.");
+
+                    if (filesFailed.Count > 0)
+                    {
+                        logger.LogError($"Unable to analyze {filesFailed.Count} {(filesFailed.Count == 1 ? "file" : "files")}: {string.Join(", ", filesFailed)}");
```

Comment:
If a template fails for at least one parameter file, that template will already be reported in the error log from AnalyzeDirectory. We could log something like "Unable to analyze {templateFilePath} with {paramsFailed} parameter file(s): {list of params file paths}", but the information will also be kind of repeated between both error logs, @anaismiller what do you think?

Reply:
Less repetition in the error log is better.

Reply:
Made a change here [`ff69599` (#297)](https://github.com/Azure/template-analyzer/pull/297/commits/ff69599852c90d50b58726f371e060cb3049666d)

The errors are still logged twice, once initially and the second time during the `Execution summary` logging, but I think that is the whole point of the `Execution summary` logging 🙂 


Logging for `analyze-template` is now:
```
Error: An exception occurred while analyzing template TemplateWithSeparateParametersFile.bicep and parameters file TemplateWithSeparateParametersFile.parameters-dev.json
Error: An exception occurred while analyzing template TemplateWithSeparateParametersFile.bicep and parameters file TemplateWithSeparateParametersFile.parameters-test.json


File: D:\Git\template-analyzer\src\Analyzer.Cli.FunctionalTests\Tests\ToTestSeparateParametersFile\TemplateWithSeparateParametersFile.bicep
Parameters File: D:\Git\template-analyzer\src\Analyzer.Cli.FunctionalTests\Tests\ToTestSeparateParametersFile\TemplateWithSeparateParametersFile.parameters.json

        Rules passed: 0

Analyzed the template with 1 parameter file.

Execution summary:
        The verbose mode (option -v or --verbose) can be used to obtain even more information about the execution.

        Summary of the errors:
                1 instance of: An exception occurred while analyzing template TemplateWithSeparateParametersFile.bicep and parameters file TemplateWithSeparateParametersFile.parameters-dev.json
                1 instance of: An exception occurred while analyzing template TemplateWithSeparateParametersFile.bicep and parameters file TemplateWithSeparateParametersFile.parameters-test.json

        0 Warnings
        2 Errors
```

Logging for `analyze-directory` is now:
```
Directory: D:\Git\template-analyzer\src\Analyzer.Cli.FunctionalTests\Tests\ToTestSeparateParametersFile\
Error: An exception occurred while analyzing template TemplateWithSeparateParametersFile.json and parameters file TemplateWithSeparateParametersFile.parameters-dev.json
Error: An exception occurred while analyzing template TemplateWithSeparateParametersFile.json and parameters file TemplateWithSeparateParametersFile.parameters-test.json


File: D:\Git\template-analyzer\src\Analyzer.Cli.FunctionalTests\Tests\ToTestSeparateParametersFile\TemplateWithSeparateParametersFile.json
Parameters File: D:\Git\template-analyzer\src\Analyzer.Cli.FunctionalTests\Tests\ToTestSeparateParametersFile\TemplateWithSeparateParametersFile.parameters.json

        Rules passed: 0
Error: An exception occurred while analyzing template TemplateWithSeparateParametersFile.bicep and parameters file TemplateWithSeparateParametersFile.parameters-dev.json
Error: An exception occurred while analyzing template TemplateWithSeparateParametersFile.bicep and parameters file TemplateWithSeparateParametersFile.parameters-test.json


File: D:\Git\template-analyzer\src\Analyzer.Cli.FunctionalTests\Tests\ToTestSeparateParametersFile\TemplateWithSeparateParametersFile.bicep
Parameters File: D:\Git\template-analyzer\src\Analyzer.Cli.FunctionalTests\Tests\ToTestSeparateParametersFile\TemplateWithSeparateParametersFile.parameters.json

        Rules passed: 0

Analyzed 2 files.

Execution summary:
        The verbose mode (option -v or --verbose) can be used to obtain even more information about the execution.

        Summary of the errors:
                1 instance of: An exception occurred while analyzing template TemplateWithSeparateParametersFile.json and parameters file TemplateWithSeparateParametersFile.parameters-dev.json
                1 instance of: An exception occurred while analyzing template TemplateWithSeparateParametersFile.json and parameters file TemplateWithSeparateParametersFile.parameters-test.json
                1 instance of: An exception occurred while analyzing template TemplateWithSeparateParametersFile.bicep and parameters file TemplateWithSeparateParametersFile.parameters-dev.json
                1 instance of: An exception occurred while analyzing template TemplateWithSeparateParametersFile.bicep and parameters file TemplateWithSeparateParametersFile.parameters-test.json

        0 Warnings
        4 Errors
```